src/dm-area/dm-map.js

```diff
@@ -673,14 +706,56 @@ export const DmMap = ({
           ...panZoomContainerStyles,
           cursor
         }}
-        onClick={ev => {
-          if (tool !== "mark") {
-            return;
+        onContextMenu={ev => {
+          if (tool === "tokens") {
+            ev.preventDefault();
+            const ref = new Referentiel(
+              panZoomRef.current.dragContainer.current
+            );
+            const [x, y] = ref.global_to_local([ev.pageX, ev.pageY]);
+            const token = [...document.querySelectorAll(".tokenCircle")].filter(
```

Comment:
I don‘t like this bail out of react solution. Can we add the event listener to the token, so we do not have to manually read from the DOM?

Reply:
I had an issue with this one for a while - basically as all the event listeners are on the PanZoom, I was not able to properly bind onClick/onContextMenu to the ObjectLayer.  Implementing #148 may make this workaround obsolete, but for the moment it was the best I could think of that still used event listeners on the PanZoom.

Reply:
Okay, I can understand this! I fixed #148 as part of #146. My current plan is to merge #146 first and then rebase your work on top of it (while adjusting the missing bits). Hopefully, that will be done until the end of the week.

Reply:
So, seeing as #146 has been merged down, shall I leave merging my changes to you @n1ru4l?

Reply:
Yes I am currently working on this!